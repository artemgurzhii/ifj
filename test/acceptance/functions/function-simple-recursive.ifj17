Function recursive (n As Integer) As Integer
dim answer as integer

if (n == 1) then
  return 1
else
  n = n - 1
  answer = recursive(n)
  answer = answer * n
end if

return answer
End Function


Scope
dim a as integer
a = recursive(a)
End Scope
