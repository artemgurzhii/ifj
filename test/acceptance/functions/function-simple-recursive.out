.IFJcode17
JUMP Scope

LABEL recursive
CREATEFRAME
DEFVAR TF@n
POPS TF@n
DEFVAR TF@answer

JUMPIFEQ RES_IF_1 TF@n int@1
JUMP RES_ELSE_1

LABEL RES_IF_1
PUSHS int@1
RETURN
JUMP END_IF_1

LABEL RES_ELSE_1
SUB TF@n TF@n int@1
PUSHS TF@n
CALL recursive
POPS TF@answer
MUL TF@answer TF@answer TF@n
JUMP END_IF_1

LABEL END_IF_1
RETURN


LABEL Scope
DEFVAR GF@a

PUSHS GF@a
CALL recursive
POPS GF@a
